model/src/main/java/org/projectnessie/api/http/HttpNamespaceApi.java

```diff
@@ -0,0 +1,131 @@
+
+@Path("namespaces")
+@Consumes(value = MediaType.APPLICATION_JSON)
+public interface HttpNamespaceApi extends NamespaceApi {
+
+  @Override
+  @PUT
+  @Path("/namespace/{ref}/{name}")
+  @Produces(MediaType.APPLICATION_JSON)
+  @Operation(summary = "Creates a Namespace")
+  @APIResponses({
+    @APIResponse(
+        responseCode = "200",
+        description = "Returned Namespace.",
+        content =
+            @Content(
+                mediaType = MediaType.APPLICATION_JSON,
+                examples = {
+                  @ExampleObject(ref = "namespace"),
+                },
+                schema = @Schema(implementation = Namespace.class))),
+    @APIResponse(responseCode = "401", description = "Invalid credentials provided"),
+    @APIResponse(responseCode = "403", description = "Not allowed to create namespace"),
+    @APIResponse(responseCode = "404", description = "Reference not found"),
+    @APIResponse(responseCode = "409", description = "Namespace already exists"),
+  })
+  Namespace createNamespace(@BeanParam @NotNull NamespaceParams params)
+      throws NessieNamespaceAlreadyExistsException, NessieReferenceNotFoundException;
+
+  @Override
+  @DELETE
+  @Path("/namespace/{ref}/{name}")
+  @Produces(MediaType.APPLICATION_JSON)
+  @Operation(summary = "Deletes a Namespace")
+  @APIResponses({
+    @APIResponse(responseCode = "200", description = "Namespace successfully deleted."),
+    @APIResponse(responseCode = "401", description = "Invalid credentials provided"),
+    @APIResponse(responseCode = "403", description = "Not allowed to delete namespace"),
+    @APIResponse(responseCode = "404", description = "Reference or Namespace not found"),
+    @APIResponse(responseCode = "409", description = "Namespace not empty"),
+  })
+  void deleteNamespace(@BeanParam @NotNull NamespaceParams params)
+      throws NessieReferenceNotFoundException, NessieNamespaceNotEmptyException,
+          NessieNamespaceNotFoundException;
+
+  @Override
+  @GET
+  @Path("/namespace/{ref}/{name}")
+  @Produces(MediaType.APPLICATION_JSON)
+  @Operation(summary = "Retrieves a Namespace")
+  @APIResponses({
+    @APIResponse(
+        responseCode = "200",
+        description = "Returned Namespace.",
+        content =
+            @Content(
+                mediaType = MediaType.APPLICATION_JSON,
+                examples = {
+                  @ExampleObject(ref = "namespace"),
+                },
+                schema = @Schema(implementation = Namespace.class))),
+    @APIResponse(responseCode = "401", description = "Invalid credentials provided"),
+    @APIResponse(responseCode = "403", description = "Not allowed to retrieve namespace"),
+    @APIResponse(responseCode = "404", description = "Reference or Namespace not found"),
+  })
+  Namespace getNamespace(@BeanParam @NotNull NamespaceParams params)
+      throws NessieNamespaceNotFoundException, NessieReferenceNotFoundException;
+
+  @Override
+  @GET
+  @Path("/{ref}")
```

Comment:
What if we have a ref named `namespace`... will the path resolution work with respect to `getNamespace`?

Reply:
yes this works. I updated one of the tests to use a Branch called `namespace`